src/prefect/filesystems.py

```diff
@@ -723,3 +725,48 @@ async def read_path(self, path: str) -> bytes:
 
     async def write_path(self, path: str, content: bytes) -> str:
         return await self.filesystem.write_path(path=path, content=content)
+
+
+class GitHub(ReadableDeploymentStorage):
```

Comment:
Should we name this more broadly to something like `GitRepository`? There's nothing in this block that is specific to GitHub.

Reply:
that's technically true, but this way we would name S3 as S3Path -- here I think we name it all based on system/service for storing files and directories; GitHub is an outlier here anyway, so this totally could be a GitRepository

Reply:
@desertaxle thinking ahead, the next change will almost certainly be to expose authentication to this block which I believe will be GitHub specific